### ISSUE TITLE
mpbridge not interrupting processing on esp32

### ISSUE THREAD

**Issue by peroveh** [AmirHmZz/mpbridge#22]
High... Excellent tool ... Have a couple of issues... When running the ESP32 with a program, in many cases it is difficult to interrupt the process for new update of python code by using "ctr alt ]". In the end i have to stop the mpbridge and erase the ESP code using esptool doing a flash erease and flash innstall of micropython. (esptool IS able to connect to the ESP32 even if mpbridge cannot)
I do not know if this is caused by me not installing with "sudo pip install -U mpbridge" in a conda environment since sudo gave an error i skipped it , using "pip install -U mpbridge" instead. 
![Screenshot from 2023-03-06 19-52-10](https://user-images.githubusercontent.com/16647600/223203853-e6626921-845e-4c08-a9b7-913c24e06a29.png)

**Comment by AmirHmZz** [comment]
Can you try running dev mode with `--auto-reset=hard` flag and share the result?

**Comment by Wind-stormger** [comment]
Have you solved your problem?

**Comment by Wind-stormger** [comment]
I suggest you use mpremote to directly connect to esp32 to check whether the REPL is normal.
```
mpremote connect /dev/ttyUSB0 repl
```

**Comment by peroveh** [comment]
Hi.. Sorry for late response on yours. 
Yes... I did get contact with "mpremote connect /dev/ttyUSB0 repl"
using ctrl-C, i got a > prompt
with a > prompt... Using ctrl-B , i got to the repl prompt >>>

I also tried the extended command: 
mpbridge dev /dev/ttyUSB0 /home/perove/Dokumenter/micropy/devroot --auto-reset=hard
This was not able to "interrupt" the processing and get me into the file update mode..

What i wanted was to get the normal update of the files (which had been modified in the PC) in a simple manner..
(from the above repl I was able to delete the boot.py which starts the normal program, so i did not have to flash the circuit this time ;-) BUT I would like a simpler update loop when experimenting with the code. ) 
How can i achieve this?

**Comment by Wind-stormger** [comment]
mpbridge dev
Can't this meet the needs you mentioned?

**Comment by peroveh** [comment]
> mpbridge dev Can't this meet the needs you mentioned?

Yes.. when it works... And for a pure repl startup it does. BUT when I start a main program from the boot.py it appears no longer to be able to enter sync mode ... The sw keeps doing its thing (sending mqtt messages over wifi) but is not able to sync a modified version of main program or boot.py.  
Even at normal repl prompt, the mpbridge dev is very slow at doing the syncing of the files...
Like now: 

```
(micropy-env) perove@perove-Swift-SF314-71:~/Dokumenter/micropy$ mpbridge dev /dev/ttyUSB0 /home/perove/Dokumenter/micropy/devroot

 - Syncing files on /dev/ttyUSB0 with /home/perove/Dokumenter/micropy/devroot
 - Entering raw repl
```

**Comment by AmirHmZz** [comment]
@peroveh Could you please try again with mpbridge `v1.4.0` ?
